core/src/toga/__init__.py

```diff
@@ -1,8 +1,15 @@
-from __future__ import annotations
+# Enable the standard library compatibility shims before doing anything else.
+#
+# __future__ imports must be at the very top of the file, and MicroPython doesn't
+# currently include a __future__ module, so this file can't contain any __future__
+# imports. Other modules imported after `compat` can use __future__ as normal.
+import sys
+
+if sys.implementation.name != "cpython":  # pragma: no cover
+    from . import compat  # noqa: F401
```

Comment:
Since there won't be any other implementations in the foreseeable future, let's cross that bridge when we come to it. I see this `compat` package as a temporary stopgap anyway – virtually all of it should eventually be upstreamed to [micropython-lib](https://github.com/micropython/micropython-lib).